pylint/utils.py

```diff
@@ -730,32 +732,93 @@ def register_messages(self, checker):
         for msgid, msg_tuple in sorted(six.iteritems(checker.msgs)):
             msg = build_message_def(checker, msgid, msg_tuple)
             # avoid duplicate / malformed ids
-            if msg.symbol in self._messages or msg.symbol in self._alternative_names:
-                raise InvalidMessageError(
-                    'Message symbol %r is already defined' % msg.symbol)
-            if chkid is not None and chkid != msg.msgid[1:3]:
-                raise InvalidMessageError(
-                    "Inconsistent checker part in message id %r (expected 'x%sxx')"
-                    % (msgid, chkid))
+            self._check_message(msg, chkid)
             chkid = msg.msgid[1:3]
             self._messages[msg.symbol] = msg
-            self._register_alternative_name(msg, msg.msgid)
+            self._register_alternative_name(msg, msg.msgid, msg.symbol)
             for old_id, old_symbol in msg.old_names:
-                self._register_alternative_name(msg, old_id)
-                self._register_alternative_name(msg, old_symbol)
+                self._register_alternative_name(msg, old_id, old_symbol)
             self._msgs_by_category[msg.msgid[0]].append(msg.msgid)
 
-    def _register_alternative_name(self, msg, name):
+    def _register_alternative_name(self, msg, msgid, symbol):
         """helper for register_message()"""
-        if name in self._messages and self._messages[name] != msg:
-            raise InvalidMessageError(
-                'Message symbol %r is already defined' % name)
-        if name in self._alternative_names and self._alternative_names[name] != msg:
-            raise InvalidMessageError(
-                'Message %s %r is already defined' % (
-                    'id' if len(name) == 5 and name[0] in MSG_TYPES else 'alternate name',
-                    name))
-        self._alternative_names[name] = msg
+        self._check_id_and_symbol_consistency(msgid, symbol)
+        self._alternative_names[msgid] = msg
+        self._alternative_names[symbol] = msg
+
+    def _check_symbol(self, msgid, symbol):
+        """ Check that a symbol is not already used. """
```

Comment:
Please can we include the parameter documentation in all of the new docstrings?

Reply:
@AWhetter what kind of parameter documentation do you want ? I did not find an example in utils.py or in your recent commit. I often use the sphinx syntax : 

    :param String msgid: should be a string of len 4, where the two first characters are the checker 
    id and the two last the message id in this checker". 

But this would be a copy pasta of the documentation of the main function called "register_message" line 722, I don't think putting it in every sub-function would be a good idea.

Reply:
Yes we use sphinx syntax. Feel free to use something shorter like "`:param str msgid: The id of the message.`" because the argument is documented fully elsewhere.

Reply:
What do you think about renaming the variable to "msg_id, msg_symbol" or "message_id" and "message_symbol" ? (So the documentation of the parameter is redondant)